### PR TITLE
Estimate time

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

**Comment by xitronix** [comment]
image

File: universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts [hunk 1]
```diff
@@ -1,18 +1,34 @@
-import {utils, providers} from 'ethers';
+import {utils} from 'ethers';
+import {handleApiResponse} from '@unilogin/commons';
+import {fetch} from '../http/fetch';
+import {GasPriceSuggestion} from '../../core/models/GasPriceSuggestion';
 
-export type GasPriceOption = 'fast' | 'cheap';
+export class GasPriceOracle {
+  private async estimateGasPrices() {
+    const response = await fetch('https://ethgasstation.info/json/ethgasAPI.json', {method: 'GET'});
```
Comment (Marik-D) [inline universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts]: Sanitize the response

File: universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+    return handleApiResponse(response);
+  }
 
-export type GasPriceSuggestion = Record<GasPriceOption, utils.BigNumber>;
+  private convertMinutesToSec(minutes: number): string {
+    return (minutes * 60).toString();
```
Comment (Marik-D) [inline universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts]: Shouldn't be in this class

Comment (Marik-D) [reply universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts]: Just make it a separate function in this file

File: universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
+  }
 
-export class GasPriceOracle {
-  constructor(private provider: providers.Provider) {
+  private convert10xGweiToWei(gwei10x: number): utils.BigNumber {
+    return utils.parseUnits((gwei10x / 10).toString(), 'gwei');
```
Comment (Marik-D) [inline universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts]: same as above

File: universal-login-sdk/src/core/models/GasPriceSuggestion.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {utils} from 'ethers';
+
+export type GasPriceOption = 'fast' | 'cheap';
+
+export type GasPriceEstimation = {
+  gasPrice: utils.BigNumber;
+  timeEstimation: string;
```
Comment (Marik-D) [inline universal-login-sdk/src/core/models/GasPriceSuggestion.ts]: Why is this a string?

File: universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts [hunk 5]
```diff
@@ -1,18 +1,37 @@
...
-export type GasPriceSuggestion = Record<GasPriceOption, utils.BigNumber>;
+import {utils} from 'ethers';
+import {handleApiResponse} from '@unilogin/commons';
+import {fetch} from '../http/fetch';
+import {GasPriceSuggestion} from '../../core/models/GasPriceSuggestion';
+import {asObject, asNumber, cast} from '@restless/sanitizers';
 
 export class GasPriceOracle {
-  constructor(private provider: providers.Provider) {
+  private async estimateGasPrices() {
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/integration/ethereum/gasPriceOracle.ts]: Rename to `fetchGasPrices`

**Pull request merged**